### PR TITLE
Feature/global 3w charts

### PR DESCRIPTION
**Pull request by frozenhelium**

### CODE REVIEW

File: src/root/views/GlobalThreeW/Charts/index.tsx [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+  key: number,
+  name: string,
+  value: number,
+}
+
+const PIE_COLORS = ['#f64752', '#fa999f', '#f87079'];
+
+
+interface PieChartProps {
+  data: any[],
```
Comment (tnagorra) [inline src/root/views/GlobalThreeW/Charts/index.tsx:29]: We shouldn't use `any` type

File: src/root/views/GlobalThreeW/Charts/index.tsx [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+        return {
+          key: datum["programme_type"],
+          name: datum["programme_type_display"],
+          value: datum["count"],
+        };
+      });
+      return dataArr;
+    }
+
+    return [];
```
Comment (tnagorra) [inline src/root/views/GlobalThreeW/Charts/index.tsx:51]: It's better to shortcircuit the if statement

File: src/root/views/GlobalThreeW/Charts/index.tsx [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+        className={styles.chartContainer}
+      >
+        <PieChart>
+          <Pie
+            data={chartData}
+            dataKey="value"
+          >
+            { data.map((entry, index) => (
+              <Cell
+                key={`cell-${index}`}
```
Comment (tnagorra) [inline src/root/views/GlobalThreeW/Charts/index.tsx:69]: It's not a good idea using index as key

File: src/root/views/GlobalThreeW/Charts/index.tsx [hunk 4]
```diff
@@ -0,0 +1,174 @@
...
+        className={styles.tick}
+      >
+        {value}
+      </text>
+    </g>
+  );
+}
+
+interface BarChartProps {
+  data: any[],
```
Comment (tnagorra) [inline src/root/views/GlobalThreeW/Charts/index.tsx:108]: We shouldn't use `any` type

File: src/root/views/GlobalThreeW/Charts/index.tsx [hunk 5]
```diff
@@ -0,0 +1,174 @@
...
+      const dataArr = data.map((datum: any) => {
+        return { key: datum[keys[0]], name: datum[keys[1]], value: datum["count"] };
+      });
+
+      return dataArr
+        .sort((a, b) => b.value - a.value)
+        .splice(0, 5); // Only show top 5
+    }
+
+    return [];
```
Comment (tnagorra) [inline src/root/views/GlobalThreeW/Charts/index.tsx:131]: Better to shortcircuit the if statement

File: src/root/views/GlobalThreeW/Charts/index.tsx [hunk 6]
```diff
@@ -0,0 +1,174 @@
...
+function GoBarChart(props: BarChartProps) {
+  const {
+    data,
+    className,
+  } = props;
+
+  const chartData = React.useMemo(() => {
+    if (data && data.length > 0) {
+      const keys = Object.keys(data[0]);
+      const dataArr = data.map((datum: any) => {
```
Comment (tnagorra) [inline src/root/views/GlobalThreeW/Charts/index.tsx:122]: The type shouldn't be `any`

File: src/root/views/GlobalThreeW/Charts/index.tsx [hunk 7]
```diff
@@ -0,0 +1,174 @@
...
+        return { key: datum[keys[0]], name: datum[keys[1]], value: datum["count"] };
+      });
+
+      return dataArr
+        .sort((a, b) => b.value - a.value)
+        .splice(0, 5); // Only show top 5
```
Comment (tnagorra) [inline src/root/views/GlobalThreeW/Charts/index.tsx:128]: Shouldn't this be configurable?

**Review by tnagorra — CHANGES_REQUESTED** [review]

**Review by barshathakuri — APPROVED** [review]

File: src/root/views/GlobalThreeW/Charts/index.tsx (hunk 7 above)
Comment (frozenhelium) [reply src/root/views/GlobalThreeW/Charts/index.tsx:128]: This is only for the Global 3W page, so I guess it's okay

**Pull request merged**